sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -116,16 +116,20 @@ object DataSourceV2Strategy extends Strategy {
            |Output: ${output.mkString(", ")}
          """.stripMargin)
 
-      val scan = DataSourceV2ScanExec(
+      val batch = scan.toBatch
+      val partitions = batch.planInputPartitions()
+      val readerFactory = batch.createReaderFactory()
+      val plan = DataSourceV2ScanExec(
```

Comment:
I mentioned this above, but I think that DataSoruceV2ScanExec only needs to be passed `output` and `batch`. That is, unless there is a benefit to calling `planInputPartitions` here, like an earlier failure?